### DISCUSSION TITLE
postgresql.conf

### DISCUSSION THREAD

**Discussion by albenard** [fail2ban/fail2ban#3660] [Q&A] [answered]
**I have a postgresql server and i find this :** 
```ini
# Fail2Ban filter for PostgreSQL authentication failures
#
# Author: Cyril Jaquier
#

[INCLUDES]
before = common.conf

[Definition]
_daemon = postgres(ql)?(-\d+(\.\d+)?)?
failregex = ^%(__prefix_line)sFATAL:  password authentication failed for user "<HOST>"\s*$
            ^%(__prefix_line)sFATAL:  no pg_hba.conf entry for host "<HOST>", user ".+", database ".+"\s*$
ignoreregex =
```

**My logfile look like this  :** 
```
2024-01-07 06:01:58.326 CET [318008] postgres@postgres FATAL:  no pg_hba.conf entry for host "47.94.144.150", user "postgres", database "postgres", no encryption
2024-01-07 06:01:58.724 CET [318009] postgres@postgres FATAL:  no pg_hba.conf entry for host "47.94.144.150", user "postgres", database "postgres", no encryption
```
with fail2ban-regex all lines for my log files are missed.
I change datepattern with the hint fail2ban-regex give me biut it's the same result.
In my postgresql configuration file I have log_line_prefix = '%m [%p] %q%u@%d '
I don't understand how configure failregex. Any biody as the same problem with the same configuration?? Thanks for advance.

**Comment by sebres** [comment, marked as answer]
Yes, this timestamp is not supported right now (because of TZ abbreviation `CET`), so `datepattern` is almost mandatory (otherwise it'd match the part without TZ, thus could result to invalid date-time, depending on current zone). If the current TZ of server is also CET (CEST in summer), then you could use standard datepattern, just then `\s[A-Z]{3,5}` must be added to `__prefix_line` RE to match TZ abbreviation that doesn't matched by `datepattern` (and so remain in the search text, because default pattern would only cut `2024-01-07 06:01:58.326` out).

But the default common prefix (`__prefix_line`) would not match the prefix in log-format in any case.

You could try to set different `__prefix_line` (then the include is not necessary anymore):
```ini
[Definition]
datepattern = ^\s*%%Y-%%m-%%d(?:T|  ?)%%H:%%M:%%S(?:\.%%f)?(?:\s*%%Z)?
__prefix_line = \s*\[\d+\] postgres@\S+\s+
failregex = ^%(__prefix_line)sFATAL:  password authentication failed for user "<ADDR>"
            ^%(__prefix_line)sFATAL:  no pg_hba.conf entry for host "<ADDR>", user "<F-USER>[^"]+</F-USER>"
```
also note the end part is rewritten (without anchor etc), because anchored it doesn't match the message anymore - e. g. due to `no encryption` inside). 

No idea whether RE `password authentication failed` would match also, because you did not provide the example log with that messages.

Alternatively you can use single failregex (also without filter, just in jail.local):
```ini
datepattern = ^\s*%%Y-%%m-%%d(?:T|  ?)%%H:%%M:%%S(?:\.%%f)?(?:\s*%%Z)?
failregex = ^\s*\[\d+\] postgres@\S+ FATAL:\s+(?:password authentication failed|no pg_hba.conf entry) for (?:user|host) "<ADDR>"
```

To test it with fail2ban-regex directly:
```
fail2ban-regex -d '^\s*%Y-%m-%d(?:T|  ?)%H:%M:%S(?:\.%f)?(?:\s*%Z)?' "$logpath" '^\s*\[\d+\] postgres@\S+ FATAL:\s+(?:password authentication failed|no pg_hba.conf entry) for (?:user|host) "<ADDR>"'
```

> In my postgresql configuration file I have log_line_prefix = '%m [%p] %q%u@%d '

Tokens like `%m` etc are allowed in `datepattern` only (let alone they need to be escaped with extra `%`) and matches parts of date or time like month, whether in all regexs one has to specify [regex tokens](https://docs.python.org/3/library/re.html#regular-expression-syntax) or variable or parameter substitution `%(var)s` or `<var>`. See [man jail.conf.5](https://manpages.debian.org/stable/fail2ban/jail.conf.5.en.html) for details.

**Comment by albenard** [comment]
Thank you this syntax is OK. Have a nice day
